tensorflow_datasets/core/image_visualizer.py

```diff
@@ -0,0 +1,129 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+
+"""Visualizer for image datasets.
+"""
+
+from absl import logging
+
+from tensorflow_datasets.core import dataset_utils
+from tensorflow_datasets.core import features as features_lib
+from tensorflow_datasets.core import lazy_imports_lib
+
+
+def match(ds_info):
+  """Checks if dataset belongs to image dataset category"""
+  image_keys = [
+      k for k, feature in ds_info.features.items()
+      if isinstance(feature, features_lib.Image)
+  ]
+
+  if not image_keys:
+    return False
+
+  sequence_keys = [
+      k for k, feature in ds_info.features.items()
+      if isinstance(feature, features_lib.Sequence)
+  ]
+
+  bbox_keys = []
+  for key in sequence_keys:
+    if isinstance(ds_info.features[key].feature, features_lib.FeaturesDict):
+      bbox_keys = [
+          k for k, feature in ds_info.features[key].items()
+          if isinstance(feature, features_lib.BBoxFeature)
+      ]
+
+  if bbox_keys:
+    return False
+
+  return True
+
+def build(ds_info, ds, **kwargs):
+  """Builds the plot for visualization"""
+  rows = kwargs.setdefault('rows', 3)
+  cols = kwargs.setdefault('cols', 3)
+  plot_scale = kwargs.setdefault('plot_scale', 3)
+  image_key = kwargs.setdefault('image_key', None)
+  args = ['rows', 'cols', 'plot_scale', 'image_key']
+
+  for key in kwargs:
+    if key not in args:
+      raise TypeError(
+          "show_example() got an unexpected keyword argument '{}'".format(key)
+          )
+
+  plt = lazy_imports_lib.lazy_imports.matplotlib.pyplot
+
+  if not image_key:
```

Comment:
Rather than duplicating the code, you should factorise in function and reuse them.
Similarly below